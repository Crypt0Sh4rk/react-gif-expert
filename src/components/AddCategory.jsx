import { useState } from "react";

import PropTypes from 'prop-types';


// export const AddCategory = ({ hookCategories }) => {
export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        // console.log(event.target.value);
        // setInputValue('Hola Mundo');
        setInputValue(event.target.value);
    };

    const onSubmit = (event) => {
        // console.log('Hola Mundo desde el submit')
        event.preventDefault(); // Evita el refresh del navegador
        // console.log(inputValue)
        if (inputValue.trim().length <= 1) return;

        // const categories = hookCategories.categories;
        // hookCategories.setCategories([ inputValue, ...categories]);

        // hookCategories.setCategories(cat => [inputValue, ...cat]);
        setInputValue('');
        onNewCategory( inputValue.trim() );
    };
    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            // onChange= { event => onInputChange(event) }
            />
        </form>
    );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}