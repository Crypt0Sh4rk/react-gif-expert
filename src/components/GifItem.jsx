import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
    // const {title, url} = image;
    return (
        <div className='card'>
            <img src={ url } alt={ title }></img>
            <p>{ title }</p>
        </div>
    )
}

/* Tarea

1. AÃ±adir PropTypes
    a. title obligatorio
    b. url obligatorio

2. Evaluar el snapshot

*/

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}