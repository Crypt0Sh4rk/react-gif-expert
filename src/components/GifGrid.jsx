// import { useState, useEffect } from "react";

import { GifItem } from "./GifItem";
import { useFetchGifs } from "../../hooks/useFetchGifs";
// import { getGifs } from "../../helpers/getGifs";

export const GifGrid = ({ category, onRemoveCategory }) => {

    // const [counter, setCounter] = useState(7)
    // const [images, setImages] = useState([]);

    // const getImages = async () => {
    //     const newImages = await getGifs(category);
    //     setImages(newImages);
    // };

    // // Llamar a getImages() cuando se crea el componente
    // useEffect(() => {
    //     // getGifs(category)
    //     //     .then( newImages => setImages(newImages));
    //     getImages();
    // }, []); // [] indica que sÃ³lo se va a disparar la primera vez que se crea y construye el componente


    // Custom hook
    const { images, isLoading } = useFetchGifs( category );

    const onClickBtnRemove = (event) => 
        onRemoveCategory(category);

    return (
        <>
            <h3>{ category }</h3>
            {
                // isLoading 
                //     ? (<h2>Cargando...</h2>)
                //     : null

                isLoading && (<h2>Cargando...</h2>)
                // <LoadingMessage isLoading={ isLoading } />
            }
            {
                !isLoading && (<button onClick={ onClickBtnRemove }>Remover</button>)
            }

            <div className="card-grid">
                {
                    // images.map(({ id, title, url }) => (
                    images.map(img => (
                        // <GifItem key={id} image={{title, url}} />
                        // <GifItem key={id} title={title} url={url} />
                        <GifItem key={img.id} {...img} /> // Esparcir propiedades
                    ))
                }
            </div>
        </>
    );
};