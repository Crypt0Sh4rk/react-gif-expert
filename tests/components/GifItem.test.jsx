import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem />', () => { 
    const title = "Superman";
    const url   = "https://one-punch.com/saitama.png";

    test('debe de hacer match con el snapshot', () => { 
        const {container} = render(
            <GifItem
                title={ title }
                url = { url }
            />
        );
        expect(container).toMatchSnapshot();
    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render( <GifItem title={title} url={url} />);
        // screen.debug();

        // console.log( screen.getByRole('img'));
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(src);
        expect(alt).toBe(title);
    })


    test('debe de mostrar el tÃ­tulo en el componente', () => { 
        render( <GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy(); // Valida que title exista como texto, y no como valor de una property
    })
})