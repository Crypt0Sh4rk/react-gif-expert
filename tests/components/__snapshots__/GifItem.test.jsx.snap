// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Superman"
      src="https://one-punch.com/saitama.png"
    />
    <p>
      Superman
    </p>
  </div>
</div>
`;
